### ISSUE TITLE
Some bugs and suggestions

### ISSUE THREAD

**Issue by Grafs** [s0ph1e/node-website-scraper#55]
1 No resources are allocated without extensions
Example:
```
<link href='https://fonts.googleapis.com/css?family=Ubuntu:400,300,400italic,500,500italic,300italic,700,700italic&subset=latin,cyrillic' rel='stylesheet' type='text/css'> 
<img alt="" src="http://1.gravatar.com/avatar/4d63e4a045c7ff22accc33dc08442f86?s=140&amp;d=%2Fwp-content%2Fuploads%2F2015%2F05%2FGood-JOb-150x150.jpg&amp;r=g" srcset="http://1.gravatar.com/avatar/4d63e4a045c7ff22accc33dc08442f86?s=280&amp;d=%2Fwp-content%2Fuploads%2F2015%2F05%2FGood-JOb-150x150.jpg&amp;r=g 2x" class="avatar avatar-140 photo avatrb" width="140" height="140">
```

2 Option in config of shutdown of control of preservation in one folder

3 Not parse Emoji
Example:
```
<img src="https://s.w.org/images/core/emoji/72x72/2764.png" alt="❤" class="emoji" draggable="false">
```
4 To make an option to specify a pause in seconds before parsing

**Comment by s0ph1e** [comment]
> 3 Not parse Emoji

After 1.0.0 released (in nearest future) you will be able to specify it in `urlFilter` option (implemented in #47)

**Comment by s0ph1e** [comment]
> 1 No resources are allocated without extensions

In version `1.0.0` missing extensions are added for html and css resources so google fonts will be saved with extension. For other types it is not possible to implement. 
You can try to write your custom `filenameGenerator` which handles gravatar images and creates correct filenames for them

> 2 Option in config of shutdown of control of preservation in one folder

I'll think about saving to existing directory #65
If it will be implemented - I think need to follow the approach which is used by `wget` - it doesn't change existing files and creates new files with different filename. So if for first page image (`http://example.com/images/bg.png`) was saved to `images/bg.png`, for second page image with the same url (`http://example.com/images/bg.png`) will be saved to `images/bg_1.png`. 
I'm not sure that it is what you want.

For now I suggest using multiple urls in `urls` option - so all pages will be saved to the same directory or download all needed pages and then merge created directories.

> 3 Not parse Emoji

As I wrote in previous comment, you can use `urlFilter` in version `1.0.0`

> 4 To make an option to specify a pause in seconds before parsing

As I understand you want to download dynamic sites. Pause will not help because scraper only handles HTTP response and doesn't execute javascript
